### PR TITLE
Mimic improvements

### PR DESCRIPTION
**Pull request by niicojs**
- fix request id so it's the same as the app
- retry with correct req id if throttled
- more accurate unknown 2 (not perfect I guess)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,12 +33,13 @@
...
     "node-pogo-protos": "^2.4.2",
-    "node-pogo-signature": "^4.0.0",
+    "node-pogo-signature": "pogosandbox/node-pogo-signature",
     "request": "^2.79.0",
     "s2-geometry": "^1.2.9"
   },
   "devDependencies": {
-    "eslint": "^3.12.2",
+    "eslint": "^3.13.1",
+    "eslint-config-airbnb-base": "^11.0.1",
```
Comment (cyraxx) [inline package.json]: Has the base config been missing before? This should however be a separate commit then.

File: package.json [hunk 2]
```diff
@@ -33,12 +33,13 @@
     "gpsoauthnode": "0.0.5",
     "long": "^3.2.0",
     "node-pogo-protos": "^2.4.2",
-    "node-pogo-signature": "^4.0.0",
+    "node-pogo-signature": "pogosandbox/node-pogo-signature",
```
Comment (cyraxx) [inline package.json]: Why the change? Is this waiting for a PR on node-pogo-signature?

File: pogobuf/lehmer.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (cyraxx) [inline pogobuf/lehmer.js]: This is a Lehmer RNG apparently. Is there an npm module for this? I'd prefer that over adding an extra file.

File: pogobuf/pogobuf.client.js [hunk 4]
```diff
@@ -944,7 +940,12 @@ function Client(options) {
             }
         }
 
-        if (!envelope.auth_ticket) {
+        let authticket = envelope.auth_ticket;
```
Comment (cyraxx) [inline pogobuf/pogobuf.client.js]: `authTicket` please

File: pogobuf/pogobuf.client.js [hunk 5]
```diff
@@ -1107,7 +1108,17 @@ function Client(options) {
...
-                        resolve(self.callRPC(requests, envelope));
+                        signedEnvelope.platform_requests = [];
+                        resolve(self.callRPC(requests, signedEnvelope));
+                        return;
+                    }
+
+                    /* Throttling, retry same request later */
+                    if (responseEnvelope.status_code === 52) {
+                        signedEnvelope.platform_requests = [];
+                        Promise.delay(2000).then(() => {
```
Comment (cyraxx) [inline pogobuf/pogobuf.client.js:1122]: Not a fan of this TBH. Can this be rewritten to use the same retry logic we already use elsewhere? In any case it should be configurable in case someone wants to roll their own throttling.

File: pogobuf/pogobuf.client.js [hunk 6]
```diff
@@ -154,12 +155,7 @@ function Client(options) {
      * @return {Promise}
      */
     this.batchCall = function() {
-        if (!self.batchRequests || !self.batchRequests.length) {
-            return Promise.resolve(false);
-        }
-
-        var p = self.callRPC(self.batchRequests);
-
+        var p = self.callRPC(self.batchRequests || []);
```
Comment (cyraxx) [inline pogobuf/pogobuf.client.js:159]: Does `self.callRPC([])` resolve to `false`? If not, then this is technically a breaking change.

**Review by cyraxx — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (niicojs) [reply package.json]: We can't find a way to reach spencer. This is just temporary, @starkevin will create a new module on rpm.
I'll update to use starkevin's repo, I guess it's better than mine (fork of a fork...)

File: package.json (hunk 1 above)
Comment (niicojs) [reply package.json]: I don't know... Did you remove airbnb base at some point?
Tell me if it should be here or not.

File: pogobuf/lehmer.js (hunk 3 above)
Comment (niicojs) [reply pogobuf/lehmer.js]: The only module I found which is close use constants different from the one used in pogo.
My code gives 100% compatibility and produce exactly the id from the actual app.

File: pogobuf/pogobuf.client.js (hunk 6 above)
Comment (niicojs) [reply pogobuf/pogobuf.client.js:159]: Currently you cannot send a request without api calls, however the app does it for its very first call.

File: pogobuf/pogobuf.client.js (hunk 4 above)
Comment (niicojs) [reply pogobuf/pogobuf.client.js]: done

File: pogobuf/pogobuf.client.js (hunk 5 above)
Comment (niicojs) [reply pogobuf/pogobuf.client.js:1122]: Actually the goal of this is not to regenerate a new requestId like before.
In the actual app in case of 52, the request **with the same id** is resend.

**Comment by starkevin** [comment]
Pointing at my repo for now is fine, as discussed we'll come up with a better way of doing the signautre and pcrypt as it's a headache at the moment

File: pogobuf/pogobuf.client.js (hunk 6 above)
Comment (cyraxx) [reply pogobuf/pogobuf.client.js:159]: Interesting. This change is fine then.

File: pogobuf/pogobuf.client.js (hunk 5 above)
Comment (cyraxx) [reply pogobuf/pogobuf.client.js:1122]: That part is fine, I was just wondering if there is a better way to handle this than to implement a fixed 2000 ms delay and retry. That's why I thought we might be able to use `bluebird-retry` like we already do elsewhere with a configurable delay and maximum number of tries.

File: pogobuf/lehmer.js (hunk 3 above)
Comment (cyraxx) [reply pogobuf/lehmer.js]: Hmmm. Maybe it should go in a new folder `lib` then. Currently in the `pogobuf` folder we only have the pogobuf files itself. Or maybe it can be merged into `pogobuf.utils.js`?

Or maybe you can just turn it into an npm module? 😃

Comment (niicojs) [reply pogobuf/lehmer.js]: I put it in utils.js

**Review by cyraxx — APPROVED** [review]
